core/src/transport/global_only.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::{
+    multiaddr::{Multiaddr, Protocol},
+    transport::{ListenerId, TransportError, TransportEvent},
+    Transport,
+};
+use log::debug;
+use std::{
+    net::{Ipv4Addr, Ipv6Addr},
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// Dropping all dial requests to non-global IP addresses.
+#[derive(Debug, Clone, Default)]
+pub struct GlobalIpOnly<T> {
```

Comment:
```suggestion
pub struct Transport<T> {
```

I'd like to eventually get all symbols in this repo to follow the convention described here: https://github.com/libp2p/rust-libp2p/issues/2217

Reply:
@thomaseizinger `Transport` is already used for the `Transport` trait in `libp2p_core::transport`. Could we not say for example `TransportGlobal` to avoid confusion?

Reply:
The idea is that users refer to it via the module name, e.g. `global_only::Transport`. I believe this is the best trade-off. For modules that contain more types, this allows users to simply import a bunch of modules from the root `libp2p` namespace and refer to symbols via `<xyz>::Transport` or `<abc>::Behaviour` and `<abc>::Event`.

We do this a lot for protocols and I think it also makes sense for transports just to be consistent.

The linked ticket also links to some more discussion if your are interested :)

Reply:
In short, what we try to balance is length of symbol name and ambiguity!

Reply:
@thomaseizinger Done